tick/inference/tests/hawkes_sumgaussians.py

```diff
@@ -0,0 +1,112 @@
+                                [-0.00341803,  0.34805652, -0.00687372]],
+                               [[-0.00341635,  0.1608013 ,  0.05531324],
+                                [-0.00342652, -0.00685425,  0.19046195]]])
+
+        np.testing.assert_array_almost_equal(learner.baseline, baseline,
+                                             decimal=6)
+        np.testing.assert_array_almost_equal(learner.amplitudes, amplitudes,
+                                             decimal=6)
+
+    def test_hawkes_sumgaussians_set_data(self):
+        """...Test set_data method of Hawkes SumGaussians
+        """
+        events = [
+            [np.array([1, 1.2, 3.4, 5.8, 10.3, 11, 13.4]),
+             np.array([2, 5, 8.3, 9.10, 15, 18, 20, 33])],
+            [np.array([2, 3.2, 11.4, 12.8, 45]),
+             np.array([2, 3, 8.8, 9, 15.3, 19])]
+        ]
+
+        learner = HawkesSumGaussians(1)
+        learner._set_data(events)
+        self.assertEqual(learner.n_nodes, 2)
+
+        events = [np.array([1, 1.2, 3.4, 5.8, 10.3, 11, 13.4]),
+                  np.array([2, 5, 8.3, 9.10, 15, 18, 20, 33])]
+
+        learner = HawkesSumGaussians(1)
+        learner._set_data(events)
+        self.assertEqual(learner.n_nodes, 2)
+
+        msg = "All realizations should have 2 nodes, but realization 1 has " \
+              "1 nodes"
+        with self.assertRaisesRegex(RuntimeError, msg):
+            events = [
+                [np.array([1, 1.2, 3.4, 5.8, 10.3, 11, 13.4]),
+                 np.array([2, 5, 8.3, 9.10, 15, 18, 20, 33])],
+                [np.array([2, 3.2, 11.4, 12.8, 45])]
+            ]
+            learner._set_data(events)
+
+    def test_hawkes_sumgaussians_parameters(self):
+        """...Test that hawkes sumgaussians parameters are correctly linked
+        """
+        learner = HawkesSumGaussians(self.int_1)
+        self.assertEqual(learner.n_gaussians, self.int_1)
+        self.assertEqual(learner._learner.get_n_gaussians(), self.int_1)
+        learner.n_gaussians = self.int_2
+        self.assertEqual(learner.n_gaussians, self.int_2)
+        self.assertEqual(learner._learner.get_n_gaussians(), self.int_2)
+
+
+        learner = HawkesSumGaussians(1, max_mean_gaussian=self.float_1)
+        self.assertEqual(learner.max_mean_gaussian, self.float_1)
+        self.assertEqual(learner._learner.get_max_mean_gaussian(), self.float_1)
+        learner.max_mean_gaussian = self.float_2
+        self.assertEqual(learner.max_mean_gaussian, self.float_2)
+        self.assertEqual(learner._learner.get_max_mean_gaussian(), self.float_2)
+
+        learner = HawkesSumGaussians(1, step_size=self.float_1)
+        self.assertEqual(learner.step_size, self.float_1)
+        self.assertEqual(learner._learner.get_step_size(), self.float_1)
+        learner.step_size = self.float_2
+        self.assertEqual(learner.step_size, self.float_2)
+        self.assertEqual(learner._learner.get_step_size(), self.float_2)
+
```

Comment:
I still think it would be a good idea :).

Testing means_gaussians and std_gaussians as well